QtCollider/widgets/QcWebView.cpp

```diff
@@ -151,25 +251,23 @@ void WebView::contextMenuEvent ( QContextMenuEvent * event )
 {
     QMenu menu;
 
-    QPoint pos = event->pos();
+  const QWebEngineContextMenuData& contextData = page()->contextMenuData();
 
-    QWebHitTestResult hitTest = page()->mainFrame()->hitTestContent( pos );
-
-    if (!hitTest.linkElement().isNull()) {
-        menu.addAction( pageAction(QWebPage::CopyLinkToClipboard) );
+  if (!contextData.linkUrl().isEmpty()) {
+      menu.addAction( pageAction(QWebEnginePage::CopyLinkToClipboard) );
         menu.addSeparator();
     }
 
-    if (hitTest.isContentEditable() || hitTest.isContentSelected()) {
-        menu.addAction( pageAction(QWebPage::Copy) );
-        if (hitTest.isContentEditable())
-            menu.addAction( pageAction(QWebPage::Paste) );
+  if (contextData.isContentEditable() || !contextData.selectedText().isEmpty()) {
+      menu.addAction( pageAction(QWebEnginePage::Copy) );
+      if (contextData.isContentEditable())
+          menu.addAction( pageAction(QWebEnginePage::Paste) );
         menu.addSeparator();
     }
 
-    menu.addAction( pageAction(QWebPage::Back) );
-    menu.addAction( pageAction(QWebPage::Forward) );
-    menu.addAction( pageAction(QWebPage::Reload) );
+  menu.addAction( pageAction(QWebEnginePage::Back) );
+  menu.addAction( pageAction(QWebEnginePage::Forward) );
+  menu.addAction( pageAction(QWebEnginePage::Reload) );
```

Comment:
this part of the file is in 4-spaces for whatever reason. consider changing it all to 2-spaces or your additions to 4-spaces